src/devices/Nmea0183/SentenceCache.cs

```diff
@@ -195,6 +273,33 @@ public bool TryGetCurrentPosition(out GeographicPosition? position, bool extrapo
             }
         }
 
+        /// <summary>
+        /// Gets the last sentence with the given id from the given talker.
+        /// </summary>
+        /// <param name="source">Source to query</param>
+        /// <param name="id">Id to query</param>
+        /// <returns>The last sentence of that type and source, null if not found</returns>
+        public NmeaSentence? GetLastSentence(string? source, SentenceId id)
+        {
+            if (source == null)
+            {
+                return GetLastSentence(id);
+            }
+
+            lock (_lock)
```

Comment:
can this cache grow indefinitely or is there an upper limit?

Reply:
Yes, this caches the last messages of each type only, and since the list of possible message types is limited (maybe 30), this cannot really grow big.